clippy_lints/src/methods/ptr_offset_with_cast.rs

```diff
@@ -0,0 +1,77 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::source::SpanRangeExt;
+use clippy_utils::sym;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_middle::ty;
+use rustc_span::Symbol;
+use std::fmt;
+
+use super::PTR_OFFSET_WITH_CAST;
+
+pub(super) fn check(cx: &LateContext<'_>, method: Symbol, expr: &Expr<'_>, recv: &Expr<'_>, arg: &Expr<'_>) {
+    let ty::RawPtr(_, _) = cx.typeck_results().expr_ty(recv).kind() else {
+        return;
+    };
+
+    let method = match method {
+        sym::offset => Method::Offset,
+        sym::wrapping_offset => Method::WrappingOffset,
+        _ => return,
+    };
+
+    // Check if the argument to the method call is a cast from usize
+    let cast_lhs_expr = match arg.kind {
+        ExprKind::Cast(lhs, _) if is_expr_ty_usize(cx, lhs) => lhs,
```

Comment:
Let's use `.is_usize()` instead of adding a utility function used only once.

```suggestion
        ExprKind::Cast(lhs, _) if cx.typeck_results().expr_ty(lhs).is_usize() => lhs,
```